perfdash/config.go

```diff
@@ -131,16 +145,30 @@ func getProwConfig() (Jobs, error) {
 	}
 	jobs := Jobs{}
 	for _, periodic := range conf.Periodics {
+		var thisPeriodicConfig Tests
 		for _, tag := range periodic.Tags {
 			if strings.HasPrefix(tag, "perfDashPrefix:") {
 				split := strings.SplitN(tag, ":", 2)
-				jobs[periodic.Name] = Tests{
-					Prefix:       strings.TrimSpace(split[1]),
-					Descriptions: performanceDescriptions,
-				}
-				break
+				thisPeriodicConfig.Prefix = strings.TrimSpace(split[1])
+				continue
 			}
+			if strings.HasPrefix(tag, "perfDashJobType:") {
+				split := strings.SplitN(tag, ":", 2)
+				jobType := strings.TrimSpace(split[1])
+				thisPeriodicConfig.Descriptions = jobTypeToDescriptions[jobType]
+				continue
+			}
+			if strings.HasPrefix(tag, "perfDash") {
+				fmt.Fprintf(os.Stderr, "warning: unknown perfdash tag name: %q\n", tag)
+			}
+		}
+		if thisPeriodicConfig.Prefix == "" && thisPeriodicConfig.Descriptions == nil {
+			continue
+		}
+		if thisPeriodicConfig.Prefix == "" || thisPeriodicConfig.Descriptions == nil {
```

Comment:
I believe we shouldn't be crashing perfdash if one job config wasn't defined properly. Let's print to stderr and proceed by neglecting it?

Reply:
This is what will happen once perfdash is up and running.
Config errors are only fatal at startup.

Reply:
But i still don't understand why do we need to return nil from this function because of 1 bad job (instead of returning the valid jobs we already find)?

Reply:
Because with the amount of garbage perfdash prints at startup, it will be impossible to notice just from a warning that there is a broken config, leading to confusion and time spent on debugging.
I prefer to fail fast rater that trying to DWIM.

Reply:
I'm still not sure if that's a good enough reason to sacrifice all our results because someone made a small spelling mistake in one of the jobs. The cost of losing all our metrics because of one such mistake (unless someone debugs it) is much higher than making it difficult to debug. Moreover, why do you say that debugging would be hard? Wouldn't it be visible that one of the jobs we were expecting the results for is missing on the dashboard, and we go back looking for issues with it. And if noone notices it, it wasn't probably worth to have the job on the dashboard in the first place :)

Reply:
Basically, if someone cares about a job, he will notice missing metrics and look for debugging it. For others, we don't want to block information of all remaining jobs.

Reply:
Three problems:
- the person who notices the lack of a graph might be someone who is not aware that this is a result of a misconfiguration, they could simply assume that such data is not available, and not escalate to perfdash owners,
- due to the amount of noise in perfdash logs, looking for the cause of a missing graph will be somewhat daunting,
- in general I really feel strongly that a program should refuse to take garbage as input, otherwise madness ensues (have you tried programming in Visual Basic? ;-) See for example how kubectl exits 1 if one of the concatenated configs on its input is invalid, but some are OK.

Reply:
> the person who notices the lack of a graph might be someone who is not aware that this is a result of a misconfiguration

I would suppose that any person with a serious purpose would know the job he wants the results for (i.e scheduler-benchmark, gce at 100-node scale, etc). So if he's trying to find it, he'll know it's missing (either because the job is missing, or its perfdash is missing - both of which should ideally lead him to escalate). This may not be true for more general purpose audience though.

Overall, your reasons do look quite reasonable to me - so I'm fine with keeping it this way.